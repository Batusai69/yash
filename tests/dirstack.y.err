pushd: --no-such-option: invalid option
Usage:  pushd [-L|-P] [dir]
pushd: +5: index out of range
popd: --no-such-option: invalid option
Usage:  popd [index]
popd: +5: index out of range
popd: directory stack is empty
dirs: --no-such-option: invalid option
Usage:  dirs [-cv] [index...]
dirs: +5: index out of range
