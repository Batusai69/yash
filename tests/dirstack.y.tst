tmp="${TESTTMP}/dirstack.y"

mkdir "$tmp"
mkdir "$tmp/1" "$tmp/2" "$tmp/3" "$tmp/4"

cd "$tmp"
pushd "$tmp/1"; >f1
pushd "$tmp/2"; >f2
pushd "$tmp/3"; >f3
pushd "$tmp/4"; >f4
echo *; popd >/dev/null
echo *; popd >/dev/null
echo *; popd >/dev/null
echo *; popd >/dev/null

if [ x"$PWD" = x"$tmp" ]; then
	echo ok
fi
echo */*
pushd 1
dirs -c
cd - >/dev/null
echo */*

echo ===== 1 =====

pushd "$tmp/1"
pushd "$tmp/2"
pushd "$tmp/3"
pushd "$tmp/4"
dirs | sed -e 's;/.*/;;'
dirs -v | sed -e 's;/.*/;;'

echo ===== 2 =====

pushd
dirs | sed -e 's;/.*/;;'
pushd
dirs | sed -e 's;/.*/;;'

echo ===== 3 =====

pushd +3
pushd -2
dirs | sed -e 's;/.*/;;'

echo ===== 4 =====

popd +1
popd -2
dirs | sed -e 's;/.*/;;'

echo ===== 5 =====

pushd +0
pushd -2
dirs | sed -e 's;/.*/;;'

echo ===== 6 =====

pushd --default-directory="$tmp/4" "$tmp/1"
pushd --default-directory="$tmp/4"
dirs | sed -e 's;/.*/;;'

echo ===== 7 =====

pushd --default-directory=+2
dirs | sed -e 's;/.*/;;'

echo ===== 8 =====

pushd "$tmp/1"
pushd "$tmp/2"
pushd "$tmp/1"
pushd "$tmp/2"
pushd --remove-duplicates "$tmp/1"
dirs | sed -e 's;/.*/;;'
echo =====
dirs -v +0 -0 | sed -e 's;/.*/;;'
dirs    +2 -2 | sed -e 's;/.*/;;'

echo ===== 9 =====

unset DIRSTACK
popd >/dev/null 2>/dev/null
echo empty popd $?

cd "$tmp"
(
cd "$tmp/1"
pushd - | sed -e 's;/.*/;;'
)
pushd --default-directory=- 2>/dev/null
echo pushd hyphen $?

pushd "$tmp/1"
readonly DIRSTACK
pushd "$tmp/2" 2>/dev/null
echo dirstack readonly $?
dirs | sed -e 's;/.*/;;'

rm -fr "$tmp"
