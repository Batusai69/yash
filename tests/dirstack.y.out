pushd no-such-option 2
pushd no-such-dir 1
pushd index out of range 1
pushd output error 0
popd no-such-option 2
popd index out of range 1
popd output error 0
popd dirstack empty 1
dirs no-such-option 2
dirs index out of range 1
dirs output error 1
===== 0 =====
1 0
2 0
3 0
4 0
5 0
6 0
f4
7 0
f3
8 0
f2
9 0
f1
10 0
ok
1/f1 2/f2 3/f3 4/f4
0
1/f1 2/f2 3/f3 4/f4
===== 1 =====
4
3
2
1
dirstack.y.tmp
+0	-4	4
+1	-3	3
+2	-2	2
+3	-1	1
+4	-0	dirstack.y.tmp
===== 2 =====
3
4
2
1
dirstack.y.tmp
4
3
2
1
dirstack.y.tmp
===== 3 =====
3
1
4
2
dirstack.y.tmp
===== 4 =====
3
2
dirstack.y.tmp
===== 5 =====
3
2
dirstack.y.tmp
===== 6 =====
4
1
3
2
dirstack.y.tmp
===== 7 =====
3
4
1
2
dirstack.y.tmp
===== 8 =====
1
2
2
3
4
2
dirstack.y.tmp
=====
+0	-6	1
+6	-0	dirstack.y.tmp
2
4
===== 9 =====
empty popd 1
dirstack.y.tmp
pushd hyphen 1
dirstack readonly 1
2
dirstack.y.tmp
